test/relay/relay.node.ts

```diff
@@ -157,32 +159,30 @@ describe('Dialing (via relay, TCP)', () => {
 
     // send an invalid relay message from the relay to the destination peer
     const connections = relayLibp2p.getConnections(dstLibp2p.peerId)
-    const stream = await connections[0].newStream(RELAY_CODEC)
-    const streamHandler = new StreamHandler({ stream })
-    streamHandler.write({
-      type: CircuitRelay.Type.STATUS
-    })
-    const res = await streamHandler.read()
-    expect(res?.code).to.equal(CircuitRelay.Status.MALFORMED_MESSAGE)
-    streamHandler.close()
+    // this should fail as the destination peer has HOP disabled
+    await expect(connections[0].newStream(RELAY_V2_HOP_CODEC))
+      .to.be.rejected()
+    // empty messages are encoded as { type: RESERVE } for the hop codec,
+    // so we make the message invalid by adding a zeroed byte
 
     // should still be connected
     const dstToRelayConn = dstLibp2p.components.connectionManager.getConnections(relayLibp2p.peerId)
     expect(dstToRelayConn).to.have.lengthOf(1)
     expect(dstToRelayConn).to.have.nested.property('[0].stat.status', 'OPEN')
   })
 
-  it('should time out when establishing a relay connection', async () => {
+  /* eslint-disable-next-line no-warning-comments */
```

Comment:
Please do not disable linting

Reply:
This is a flaky test that I'm trying to solve. Would it be okay to leave the TODO comments in this case?